### PR TITLE
Finish http implementation :)

### PR DESCRIPTION
**Pull request by pedrovgs**
With love from Karumi's HQs

### CODE REVIEW

File: src/main/java/com/karumi/reddo/view/MatrixLedView.java [hunk 1]
```diff
@@ -35,40 +34,41 @@ public void showMessages(List<String> messages) {
...
+      Log.e(e.toString());
     }
   }
 
-  private void drawFrame(BufferedImage outputImage, DataOutputStream stream) throws IOException {
-    stream.writeInt(outputImage.getHeight());
-    stream.writeInt(outputImage.getWidth());
-
+  private String generateBase64Image(BufferedImage outputImage) {
+    byte[] data = new byte[3 * outputImage.getWidth() * outputImage.getHeight()];
```
Comment (Serchinastico) [inline src/main/java/com/karumi/reddo/view/MatrixLedView.java]: I know it's my code but I'd give a name to that 3, something like COLOR_COUNT

File: src/main/java/com/karumi/reddo/view/MatrixLedView.java [hunk 2]
```diff
@@ -35,40 +34,41 @@ public void showMessages(List<String> messages) {
...
+        data[x * 3 + y * (3 * outputImage.getWidth())] = (byte) (rgb & 0xFF0000);
+        data[x * 3 + y * (3 * outputImage.getWidth()) + 1] = (byte) (rgb & 0x00FF00);
+        data[x * 3 + y * (3 * outputImage.getWidth()) + 2] = (byte) (rgb & 0x0000FF);
       }
     }
+    return Base64.getEncoder().encodeToString(data);
   }
 
-  private void waitForNextFrame() {
+  private void waitForNextFrame(int imageWidth) {
```
Comment (Serchinastico) [inline src/main/java/com/karumi/reddo/view/MatrixLedView.java]: This is the minimum time required to paint the message on the led, if we add network latencies, IO processing and whatsoever to the equation then we are probably going to be sending messages a bit too fast and at some point (probably days!) the queue will be full and will drop some requests. 
We obviously need another endpoint in the raspberry to retrieve the estimated time of execution for all the enqueued tasks and avoid sending anything until that time passes.

... or we could just let it be :joy:

**Comment by Serchinastico** [comment]
:+1:

File: src/main/java/com/karumi/reddo/view/MatrixLedView.java (hunk 1 above)
Comment (pedrovgs) [inline src/main/java/com/karumi/reddo/view/MatrixLedView.java]: Fixed!

File: src/main/java/com/karumi/reddo/view/MatrixLedView.java (hunk 2 above)
Comment (pedrovgs) [inline src/main/java/com/karumi/reddo/view/MatrixLedView.java]: Let It GO!

**Pull request merged**